packages/ui/test/working-groups/modals/AddWorkerStakeModal.test.tsx

```diff
@@ -0,0 +1,194 @@
+      expect(submitButton).not.toBeDisabled()
+    })
+
+    it('Too low stake', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '100' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+
+    it('Stake exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '10000' } })
+
+      expect(submitButton).toBeDisabled()
+    })
+  })
+
+  describe('Sign step', () => {
+    it('Goes to sign modal', async () => {
+      const { queryByText } = await renderModalAndProceedToSign()
+
+      expect(queryByText('Authorize transaction')).toBeDefined()
+    })
+
+    it('Transaction exceed balance', async () => {
+      renderModal()
+
+      const input = await getStakeInput()
+      const submitButton = await getButton('Add Stake')
+
+      fireEvent.change(input, { target: { value: '1000' } })
+      submitButton.click()
+
+      const signButton = await getButton('Sign transaction and Stake')
+
+      expect(signButton).toBeDisabled()
+    })
+
+    describe('Pending', () => {
+      it('Renders wait for transaction', async () => {
+        stubTransactionPending(transfer)
+
+        await renderModalAndProceedToSign(true)
+
+        expect(screen.queryByText('Pending transaction')).toBeDefined()
+      })
+    })
+
+    describe('Success', () => {
+      beforeEach(async () => {
+        stubTransactionSuccess(transfer, 'storageWorkingGroup', 'StakeIncreased')
+      })
+
+      it('Renders transaction success', async () => {
+        await renderModalAndProceedToSign(true)
+
+        expect(await screen.findByText('Success')).toBeDefined()
+      })
+    })
+
+    describe('Failure', () => {
+      it('Renders transaction failure', async () => {
+        stubTransactionFailure(transfer)
+        await renderModalAndProceedToSign(true)
+
+        expect(await screen.queryByText('Failure')).toBeDefined()
+      })
+    })
+  })
+
+  const renderModalAndProceedToSign = async (shouldSign = false) => {
+    const modal = renderModal()
+
+    const submitButton = await getButton('Add Stake')
+    submitButton.click()
+
+    if (shouldSign) {
+      const signButton = await getButton('Sign transaction and Stake')
+      signButton.click()
+    }
+
+    return modal
+  }
+
+  const getStakeInput = () => {
+    return screen.getByTestId('amount-input')
+  }
+
+  function renderModal() {
+    return render(
+      <MockKeyringProvider>
+        <ApiContext.Provider value={api}>
+          <MockQueryNodeProviders>
+            <ModalContext.Provider value={mockModalContext}>
+              <AddWorkerStakeModal />
+            </ModalContext.Provider>
+          </MockQueryNodeProviders>
+        </ApiContext.Provider>
+      </MockKeyringProvider>
+    )
```

Comment:
```suggestion
   render(
      <MockKeyringProvider>
        <ApiContext.Provider value={api}>
          <MockQueryNodeProviders>
            <ModalContext.Provider value={mockModalContext}>
              <AddWorkerStakeModal />
            </ModalContext.Provider>
          </MockQueryNodeProviders>
        </ApiContext.Provider>
      </MockKeyringProvider>
    )
```
If the only reason to return render result is to get `get|find|queryBy*()` query, I think that is better to use `screen.get|find|queryBy*()`.